src/v3.rs

```diff
@@ -339,11 +340,45 @@ impl Personalization {
 
     /// Add a dynamic template data field.
     pub fn add_dynamic_template_data(mut self, dynamic_template_data: SGMap) -> Personalization {
-        self.dynamic_template_data
-            .get_or_insert_with(|| SGMap::with_capacity(dynamic_template_data.len()))
-            .extend(dynamic_template_data);
+        // We can safely unwrap & unreachable here since SGMap will always serialize
+        // to a JSON object.
+        let mut new_vals = match to_value(dynamic_template_data).unwrap() {
+            Object(map) => map,
+            _ => unreachable!(),
+        };
+        if let Some(mut old_vals) = self.dynamic_template_data {
```

Comment:
I think we can use `get_or_insert_with` here still to cut down on the branching.